libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -3235,6 +3142,122 @@ string YulUtilFunctions::conversionFunction(Type const& _from, Type const& _to)
 	});
 }
 
+string YulUtilFunctions::copyStructToStorageFunction(StructType const& _from, StructType const& _to)
+{
+	solAssert(_to.dataStoredIn(DataLocation::Storage), "");
+	solAssert(_from.structDefinition() == _to.structDefinition(), "");
+
+	string functionName =
+		"copy_struct_to_storage_from_" +
+		_from.identifier() +
```

Comment:
Can this cause collisions in case there are two structs from difference sources with the same name? I never remember whether `identifier` is fully qualified

Reply:
For struct it uses
`return "t_struct" + parenthesizeUserIdentifier(m_struct.name()) + to_string(m_struct.id()) + identifierLocationSuffix();`
so should be fine

Reply:
Ok!